training/pysalmcount/pysalmcount/motion_detect_stream.py

```diff
@@ -89,6 +103,16 @@ def run(self):
 
         out.release()
 
+        metadata = utils.get_video_metadata(filename)
+        if metadata is not None:
+            logger.info(f"Metadata for video file {filename}: {metadata}")
+            metadata_filepath = VideoSaver.filename_to_metadata_filepath(Path(filename))
+            with open(str(metadata_filepath), 'w') as f:
+                json.dump(asdict(metadata), f)
+            logger.info(f"Saving metadata file to harddrive: {str(metadata_filepath)}")
```

Comment:
How about we move this logger line just before the opening of the file for writing? That way if there's an error the log first shows the context of what's happening.